integration-tests/utils/integration-helpers.js

```diff
@@ -117,34 +117,33 @@ const withBilling = test => async options => {
 
 const trimClustersOldestFirst = _.flow(
   _.remove({ status: 'Deleting' }),
-  _.remove({ status: 'Creating' }),
   _.sortBy('createdDate')
 )
 
 const currentCluster = _.flow(trimClustersOldestFirst, _.last)
 
-const getCurrentCluster = withUserToken(async ({ billingProject, context, testUrl, token }) => {
+const getCurrentCluster = withUserToken(async ({ billingProject, context, email, testUrl, token }) => {
   const ajaxPage = await context.newPage()
   await ajaxPage.goto(testUrl)
   await signIntoTerra(ajaxPage, token)
 
-  const clusters = await ajaxPage.evaluate(billingProject => {
-    return window.Ajax().Clusters.list({ googleProject: billingProject })
-  }, billingProject)
+  const clusters = await ajaxPage.evaluate((billingProject, email) => {
+    return window.Ajax().Clusters.list({ googleProject: billingProject, creator: email })
+  }, billingProject, email)
 
   await ajaxPage.close()
   return currentCluster(clusters)
 })
 
-const deleteCluster = withUserToken(async ({ billingProject, context, testUrl, token }) => {
+const deleteCluster = withUserToken(async ({ billingProject, context, email, testUrl, token }) => {
```

Comment:
Since this is using `withUserToken`, it's using Scarlet Flowerpicker's token to find and delete the cluster. Do we want to continue doing that? I can see arguments both ways (see other comment as well).

Reply:
I could go either way. I'm inclined to leave it as is for now, but I'm curious about your arguments. Either choice has the advantage of possibly exposing a bug in one of two code paths in leonardo. Using the user token would avoid the potential bug of deleting the wrong clusters. However, we should probably do some clean-up/refactoring to make it clear what tokens we're using where. I was just looking and found it really hard to trace... and I think we're even switching up which token is being passed through without calling it anything different than `token`. I'd rather we take that on in a separate PR and not muddy it further in this PR.

Reply:
I was initially surprised that the deletion was being done by a different user than the creation. But after thinking about this more as a general cleanup step, it doesn't seem that unreasonable, since S.F. is essentially an admin user.

I agree that there's some cleanup and clarifying we could do here, and I also agree that it doesn't need to go in this PR.